protocol/wlr-layer-shell-unstable-v1.xml

```diff
@@ -0,0 +1,279 @@
+
+        Clients should arrange their surface for the new states, and then send
+        an ack_configure request with the serial sent in this configure event at
+        some point before committing the new surface.
+
+        The client is free to dismiss all but the last configure event it
+        received.
+
+        The width and height arguments specify the size of the window in
+        surface-local coordinates.
+
+        The size is a hint, in the sense that the client is free to ignore it if
+        it doesn't resize, pick a smaller size (to satisfy aspect ratio or
+        resize in steps of NxM pixels). If the client picks a smaller size and
+        is anchored to two opposite anchors (e.g. 'top' and 'bottom'), the
+        surface will be centered on this axis.
+
+        If the width or height arguments are zero, it means the client should
+        decide its own window dimension.
+      </description>
+      <arg name="serial" type="uint"/>
+      <arg name="width" type="uint"/>
+      <arg name="height" type="uint"/>
+    </event>
+
+    <enum name="error">
+      <entry name="invalid_input_device" value="0" summary="input device bitfield is invalid"/>
+      <entry name="invalid_anchor" value="1" summary="anchor bitfield is invalid"/>
+      <entry name="invalid_surface_state" value="2"
+	     summary="the client provided an invalid surface state"/>
+    </enum>
+
+    <enum name="anchor" bitfield="true">
+      <entry name="top" value="1" summary="the top edge of the anchor rectangle"/>
+      <entry name="bottom" value="2" summary="the bottom edge of the anchor rectangle"/>
+      <entry name="left" value="4" summary="the left edge of the anchor rectangle"/>
+      <entry name="right" value="8" summary="the right edge of the anchor rectangle"/>
+    </enum>
+  </interface>
+
+  <interface name="zwlr_layer_input_v1" version="1">
+    <description summary="layer surface input configuration object">
+      Clients can use this interface to specify input semantics for a layer
+      surface on a given seat. By default, layer surfaces are considered
+      non-interactive by seats, and will not participate in their focus
+      semantics or receive input events for them.
+
+      Input state is double-buffered.  Protocol requests modify the pending
+      state, as opposed to the current state in use by the compositor. The
+      wl_surface.commit request for the associated layer surface atomically
+      applies all pending state, replacing the current state. After commit, the
+      new pending state is as documented for each related request.
+    </description>
+
+    <request name="set_events">
+      <description summary="set input events to receive">
+        Requests that the seat send input events for the specified input devices
+        to this layer surface.
+
+        Positional events (pointer and touch) will only be sent if the layer
+        surface is the top-most interactive surface, and only when the position
```

Comment:
What does "only when the position of these events are relative to the surface" mean?

Reply:
The phrasing here is kind of awkward, I should change it. It's just reminding the reader that wl_pointer events are always relative to a surface, not in absolute coordinates.

Reply:
The protocol wording for this is usually "in surface-local coordinates".